### PR TITLE
[ENG-3649] Fileslist a11y fixes

### PR DESCRIPTION
**Pull request by futa-ikeda**
-   Ticket: [ENG-3649]
-   Feature flag: n/a

## Purpose
- Fix accessibility issues on registries file list page

## Summary of Changes
- Remove uses of nested interaction elements
  - Add styling for `<ResponsiveDropdown>` to make it look like a `<Button>`
- Remove some uses of `outline: none/0` to let keyboard users know where they are on the page
- Add aria-label to Filter textbox

## Screenshot(s)
NA
## Side Effects
None
## QA Notes
- These changes should fix the existing A11y issues on this page

[ENG-3649]: https://openscience.atlassian.net/browse/ENG-3649?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/osf-components/addon/components/file-actions-menu/template.hbs [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
+        aria-label={{t 'file_actions_menu.actions' filename=@item.name}} 
         data-test-file-download-share-trigger
         data-analytics-name='File actions'
     >
-        <Button
-            aria-label={{t 'file_actions_menu.actions' filename=@item.name}} 
-        >
-            <FaIcon @icon='ellipsis-v' />
-        </Button>
+        <FaIcon @icon='ellipsis-v' />
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/file-actions-menu/template.hbs:12]: How about Lines 29 and 49 below, the triggers for the embed menu and the sharing buttons? Do those need to also not be buttons?

File: lib/osf-components/addon/components/file-browser/template.hbs [hunk 2]
```diff
@@ -3,32 +3,33 @@
...
     <div local-class='OptionBar__left {{if this.isMobile 'Mobile'}}'>
         <Input
+            aria-label={{t 'registries.overview.files.filter_label'}}
             data-test-file-search
             @type='text'
             class='form-control'
             placeholder='{{t 'general.filter'}}'
             @value={{readonly @manager.filter}}
-            {{on 'keyup' this.handleInput}}
+            {{on 'input' this.handleInput}}
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/file-browser/template.hbs:12]: Is this behaving properly, or is it missing the last input still? Easiest to see when you clear out the contents of the filter.

File: lib/osf-components/addon/components/responsive-dropdown/template.hbs [hunk 3]
```diff
@@ -9,7 +9,7 @@
     as |dd|
 >
     {{yield (hash
-        trigger=(component dd.Trigger defaultClass=(local-class 'Trigger'))
+        trigger=(component dd.Trigger defaultClass=(local-class (if @buttonStyling 'Button')))
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/responsive-dropdown/template.hbs:12]: Do you need an else 'Trigger' on this styling? It looks like you're losing the defaultClass you had before.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Looks like you might have missed a couple of triggers, plus there are a couple of small questions.

File: lib/osf-components/addon/components/file-actions-menu/template.hbs (hunk 1 above)
Comment (futa-ikeda) [reply lib/osf-components/addon/components/file-actions-menu/template.hbs:12]: Totally missed those!

File: lib/osf-components/addon/components/file-browser/template.hbs (hunk 2 above)
Comment (futa-ikeda) [reply lib/osf-components/addon/components/file-browser/template.hbs:12]: Yea, this one works as expected! Clearing the filter textbox will re-fetch all the files, typing in `Until` will return no results when I only have files called `UntitledX`, etc

File: lib/osf-components/addon/components/responsive-dropdown/template.hbs (hunk 3 above)
Comment (futa-ikeda) [reply lib/osf-components/addon/components/responsive-dropdown/template.hbs:12]: The `.Trigger` style was only there to add `outline: 0` onfocus, so I just removed it completely

**Review by brianjgeiger — APPROVED** [review]
Looks good except for the merge conflict.

**Pull request merged**